Mathlib/MeasureTheory/Integral/Layercake.lean

```diff
@@ -268,8 +282,9 @@ variable {β : Type*} [MeasurableSpace β] [MeasurableSingletonClass β]
 
 namespace Measure
 
-theorem meas_le_ne_meas_lt_subset_meas_pos {R : Type*} [LinearOrder R] [MeasurableSpace R]
-    [MeasurableSingletonClass R] {g : α → R} (g_mble : Measurable g) {t : R}
+theorem meas_le_ne_meas_lt_subset_meas_pos₀
+    {α : Type*} [MeasurableSpace α] {μ : Measure α} {R : Type*} [LinearOrder R] [MeasurableSpace R]
+    [MeasurableSingletonClass R] {g : α → R} (g_mble : NullMeasurable g μ) {t : R}
```

Comment:
do you really need measurability assumptions here? You always have `μ {a : α | t < g a} ≤ μ {a : α | t ≤ g a}` by monotonicity, and `μ {a : α | t ≤ g a} ≤ μ {a : α | t < g a} + μ {a : α | t = g a}` by `measure_union_le`, so if `μ {a : α | t = g a}` vanishes you deduce that the two measures coincide, contradicting `ht`.

Reply:
Thank you, measurability assumptions are indeed not needed! I still find it surprising that lemmas of this sort can hold without measurability assumptions, but this is a substantial improvement.

The naming was also strange (probably a remnant/combination of earlier phrasings); I changed it to `meas_eq_pos_of_meas_le_ne_meas_lt`.